Dockerfile

```diff
@@ -1,41 +1,64 @@
-FROM debian:bullseye-slim as prepared-build-image
+ARG BASE_IMAGE=debian:bullseye-slim
+ARG GIT_REPO=https://github.com/snowblossomcoin/snowblossom.git
+ARG GIT_REF=master
+
+
+# prepares a build image that caches all bazel packages
+# so they don't have to be redownloaded every commit
+FROM $BASE_IMAGE as prepared-build-image
+ARG GIT_REPO
+ARG GIT_REF
 ARG DEBIAN_FRONTEND=noninteractive
 # mkdir line debian/openjdk bug https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863199#23
 RUN \
-mkdir -p /usr/share/man/man1 && \
-apt-get update -q && \
-apt-get install -yqq apt-utils 2>&1 && \
-apt-get install -qq --no-install-suggests --no-install-recommends git bazel-bootstrap && \
-apt-get clean
+  mkdir -p /usr/share/man/man1 \
+  && apt-get update -q \
+  && apt-get install -yqq apt-utils 2>&1 \
+  && apt-get install -qq --no-install-suggests --no-install-recommends \
+    git \
+    bazel-bootstrap \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
 RUN \
-git clone --depth 1 --branch master https://github.com/snowblossomcoin/snowblossom.git /snowblossom && \
-cd /snowblossom && \
-bazel build :Everything_deploy.jar
+  git clone --depth 1 --branch "${GIT_REF}" "${GIT_REPO}" /snowblossom \
```

Comment:
As the default input is `master`, this will be cached and not refresh the base, if master moves.

You'd need to pass an explicit hash in every time in the cronjob above.

What does `${GITHUB_REF##*/}` actually expand to on a timed build of `master`?

Reply:
```GITHUB_REF | The branch or tag ref that triggered the workflow. For example, refs/heads/feature-branch-1. If neither a branch or tag is available for the event type, the variable will not exist.```

https://docs.github.com/en/actions/reference/environment-variables

---

```

`${refs/heads/feature-branch-1##*/}` expands to `feature-branch-1`